EDU-709 - splitting export page (#3042)
temporalio/documentation: docs/production-deployment/cloud/aws-export-s3.mdx

> ---

Reviewer: There's something going on with an anchor. Once that's fixed, it should deploy to Vercel so we can get a preview. 

"Linking to #verify (resolved as: /cloud/export/aws-export-s3#verify)"

Author: Fixed in a later commit.

id: aws-export-s3
title: Exporting Workflow Event History to AWS S3
sidebar_label: AWS Export to S3
description: Export Workflow History to AWS S3
slug: /cloud/export/aws-export-s3
toc_max_heading_level: 4
keywords:
  - explanation
  - how-to
  - operations
  - temporal cloud
  - term
  - verify
  - workflow history export
  - aws
tags:
  - explanation
  - how-to
  - operations
  - temporal-cloud
  - term
  - verify
  - workflow-history-export
  - aws
---

> ### Prerequisites {#prerequisites}

Reviewer: I might check with other pages but I would make this level H2 header since it looks a bit odd if you click the gcp one and its not consistent. I think the gcp one is larger.

Author: Sounds good. Copy pasta error

Before configuring the Export Sink, ensure you have the following:

1. An AWS Account with write permission to an S3 bucket.
2. An AWS S3 bucket.
   1. The S3 bucket must reside in the same region as your Namespace.
3. (optional) A KMS ARN associated with the S3 bucket.

## Configure Workflow History Export 

> **How to configure Workflow History Export**

Reviewer: How to Configure Workflow History Export Using Temporal Cloud UI or tcld

> You can use either the [Temporal Cloud UI](#using-temporal-cloud-ui) or [tcld](#using-tcld) to configure the Workflow History Export.

Reviewer 2: ```suggestion
You can use either the [Temporal Cloud UI](#using-temporal-cloud-ui) or [`tcld`](#using-tcld) to configure the Workflow History Export.
```

The Temporal Cloud UI provides two ways for configuring Workflow History Export:

- [Automated setup](#automated-setup) (recommended): The Cloud UI launches the AWS CloudFormation Console to create a stack, with write permission to the S3 bucket.
- [Manual setup](#manual-setup): The Cloud UI provides a CloudFormation template for users to manually configure a CloudFormation stack.

### Using Temporal Cloud UI

> The following steps guide you through setting up Workflow History Export using the Temporal Cloud UI.

Reviewer: This section seems to be about automated and manual setup from my understanding. I would say that here in this sentence and give a table of contents so people can easily click to it.

![](/img/export-sink-ui.png)

:::tip

> Don't forget to click Create at the end of your setup to confirm your export.

Reviewer: I would make Create bold since it looks a little odd on the preview

Author: Thanks! Fixed.

:::

#### Automated setup

> The automated setup creates a CloudFormation stack with write permission to the S3 bucket.

Reviewer: ```suggestion
You can use the automated setup to create a CloudFormation stack with write permission to your S3 bucket.
```
for active voice

> [Verify the export setup](#verify) before saving the configuration.

Reviewer: this anchor seems to be problem

Reviewer: ```suggestion 
Make sure to verify the export setup before you save the configuration.
```
To read more easily.

1. Open the Temporal Cloud UI and navigate to the Namespace you want to configure.
2. Select **Configure** from the **Export** card.
3. Provide the following information to configure the export sink and then select **Create and launch stack**:

>    1. Name: A name for the export sink.

Reviewer: These can be dashes. I don't believe we usually do numericals after a numerical.

Author: Fixed!

   2. AWS S3 Bucket Name: The name of the configured AWS S3 bucket to send Closed Workflow Histories to.
   3. AWS Account ID: The AWS account ID.
   4. Role Name: The name of the AWS IAM role to use for the CloudFormation stack that has write permission to the S3 bucket.
   5. KMS ARN: (optional) The ARN of the AWS KMS key to use for encryption of the exported Workflow History.
4. You will be taken to the CloudFormation Console to create the stack with pre-populated information.
   1. Review the information and then select **Create stack**.

#### Manual setup

> The manual setup provides a CloudFormation template to manually configure a CloudFormation stack.

Reviewer: ```suggestion
You can manually configure a CloudFormation stack using the provided template.
```
for active voice

1. Open the Temporal Cloud UI and navigate to the Namespace you want to configure.
2. Select **Configure** from the **Export** card.
3. Select **Manual** from **Access method**.
   - Enter the Template URL into your web browser to download your copy of the CloudFormation template.
   - Configure the CloudFormation template for your export sink.
   - Follow the steps in the [AWS documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-using-console-create-stack-template.html) by uploading the template to the CloudFormation console.

> ### Using tcld

Reviewer 2: ```suggestion
### Using `tcld`
```
